src/wallet/wallet.cpp

```diff
@@ -1444,13 +1444,26 @@ void CWalletTx::GetAmounts(std::list<COutputEntry>& listReceived,
 
     // Compute fee:
     CAmount nDebit = GetDebit(filter);
-    if (nDebit > 0) // debit>0 means we signed/sent this transaction
-    {
+    bool isFromMyTaddr = nDebit > 0; // debit>0 means we signed/sent this transaction
+
+    if (isFromMyTaddr) {// debit>0 means we signed/sent this transaction
         CAmount nValueOut = GetValueOut(); // transasparent outputs plus the negative Sapling valueBalance
         CAmount nValueIn = GetShieldedValueIn();
         nFee = nDebit - nValueOut + nValueIn;
     }
 
+    // If we sent utxos from this transaction, create output for value taken from (negative valueBalance)
+    // or added (positive valueBalance) to the transparent value pool by Sapling shielding and unshielding.
+    if (isFromMyTaddr) {
+        if (sapData->valueBalance < 0) {
```

Comment:
(a778e39eafc78f772210c38600e3935f8c5ea5ae)
`sapData` could be nullopt here.
Also this can be attached to the previous `if (isFromMyTaddr) {}` block.